docs/src/main/asciidoc/kubernetes-dev-services.adoc

```diff
@@ -41,6 +42,99 @@ The default service name is `kubernetes`.
 Sharing is enabled by default in dev mode, but disabled in test mode.
 You can disable the sharing with `quarkus.kubernetes-client.devservices.shared=false`.
 
+== What else for the developers
+
+If you would like to develop test cases running top of the kubernetes cluster (launched as test container by the Dev Service), then add the following dependencies to your pom file
+
+[source,xml,role="primary asciidoc-tabs-target-sync-cli asciidoc-tabs-target-sync-maven"]
+.pom.xml
+----
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-kubernetes-client</artifactId>
+</dependency>
+<dependency>
+     <groupId>io.quarkus</groupId>
+     <artifactId>quarkus-kubernetes-client-spi</artifactId>
+</dependency>
+----
+
+and set the Quarkus properties to select the flavor, or kube version.
+
+Then you will be able to create a Fabric8 Kubernetes Client object able to perform many kube tasks as detailed part of this https://github.com/fabric8io/kubernetes-client/blob/main/doc/CHEATSHEET.md[cheat sheet].
+
+.Simple Bean Example
+[source,java]
+----
+package org.acme;
+
+import org.junit.jupiter.api.Test;
+
+import io.fabric8.kubernetes.api.model.*;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.quarkus.test.junit.QuarkusTest;
+
+@QuarkusTest
+public class ArgocdExtensionDevModeTest {
+
+    @Inject
+    private KubernetesClient client;
+
+    @Test
+    public void testCreatePod() {
+        client.resource(new PodBuilder()
+           .withMetadata(<METADATA_OBJECT>)
+           .withSpec(<SPEC_OBJECT>)
+           .build())
+           .inNamespace(<USER_NAMESPACE>)
+           .create();
+    }
+----
+<1> Quarkus kubernetes client property of the KubeConfig YAML string populated by the kind test container
+
+For the folks interested to develop a new `DevService` extension able to rely on a kind container in order to customize the cluster and to deploy by example: Argocd, Tekton, Certificate Manager, Keycloak etc. then you can create an extension project, add the following dependency to your `deployment` maven module:
+
+[source,xml,role="primary asciidoc-tabs-target-sync-cli asciidoc-tabs-target-sync-maven"]
+.pom.xml
+----
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-kubernetes-client-deployment</artifactId>
+</dependency>
+----
+
+create a DevService `Processor` class where you include the parameters: `KubernetesDevServiceRequestBuildItem` and `KubernetesDevServiceInfoBuildItem` within the body of the method which has been annotated with `@BuildStep`
+
+.DevService Extension using Kube
+[source,java]
```

Comment:
It's always a good idea to explain why we are presenting code before hand. In this case there is no intro, no explanation on why and when such code is needed.

Reply:
I removed the section for the extension's developers